lib/delivery-locations-resolver.js

```diff
@@ -160,24 +167,118 @@ class DeliveryLocationsResolver {
     })
   }
 
+  static extractLocationCode (item) {
+    try {
+      return item.holdingLocation[0].id.split(':').pop()
+    } catch (e) {
+      logger.warn('There is an item in the index with missing or malformed holdingLocation', item)
+    }
+  }
+
+  static formatLocations (locations, deliveryLocationTypes) {
+    if (!locations) return []
+    return (
+      // Filter out anything not matching the specified deliveryLocationType
+      locations.filter((location) => {
+        if (deliveryLocationTypes) {
+          return location.deliveryLocationTypes &&
+            arrayIntersection(location.deliveryLocationTypes, deliveryLocationTypes).length > 0
+        } else return true
+      })
+        // Format locations in the manner api consumers expect
+        .map((location) => {
+          return {
+            id: `loc:${location.code}`,
+            label: location.label
+          }
+        })
+        // Either way, sort deliveryLocation entries by name:
+        .sort((l1, l2) => {
+          if (l1.label < l2.label) return -1
+          return 1
+        })
+    )
+  }
+  static extractRecapBarcodes (items) {
+    return items
+      .filter((item) => {
+        // It's in ReCAP if 1) it has a RC location
+        return itemHasRecapHoldingLocation(item) ||
+          // .. or 2) it's a partner item:
+          !isItemNyplOwned(item)
+      })
+      .map(barcodeFromItem)
+  }
+
+  static attachRecapDeliveryInfo (item) {
+    const info = this.getRecapDeliveryInfo(item)
+    item.eddRequestable = info.eddRequestable
+    item.deliveryLocation = info.deliveryLocation
+    return item
+  }
+
+  static attachOnsiteDeliveryInfo (item) {
+    const info = this.getOnsiteDeliveryInfo(item)
+    item.eddRequestable = info.eddRequestable
+    item.deliveryLocation = info.deliveryLocation
+    return item
+  }
+
+  static getRecapDeliveryInfo (item) {
+    // Delivery locations are not valid if [NYPL] item's holding
+    // location is not requestable:
+    let deliveryLocation
+    let eddRequestable
+    if (this.requestableBasedOnHoldingLocation(item) || !isItemNyplOwned(item)) {
+      deliveryLocation = this.deliveryLocationsByRecapCustomerCode(item.recapCustomerCode[0])
+      eddRequestable = this.__eddRequestableByCustomerCode(item.recapCustomerCode[0])
+    } else {
+      deliveryLocation = []
+      eddRequestable = false
+    }
+    return { deliveryLocation: deliveryLocation, eddRequestable: eddRequestable }
+  }
+
+  static getOnsiteDeliveryInfo (item) {
+    let deliveryInfo = {}
+    const holdingLocationCode = this.extractLocationCode(item)
+    const sierraData = sierraLocations[holdingLocationCode]
+    if (!sierraData) {
+      // This case is mainly to satisfy a test which wants eddRequestable = false
+      // for a made up location code.
+      logger.warn('Malformed or missing holding location code: ', holdingLocationCode)
+      deliveryInfo.eddRequestable = false
+      deliveryInfo.deliveryLocation = []
```

Comment:
May be better to initialize these defaults up at line 243? (i.e. default is eddRequestable false and empty deliveryLocations)